homeassistant/components/sensibo/climate.py

```diff
@@ -242,27 +259,37 @@ def target_temperature_step(self) -> float | None:
     @property
     def fan_mode(self) -> str | None:
         """Return the fan setting."""
-        fan_mode: str | None = self.device_data.fan_mode
-        return fan_mode
+        if (fan_mode := self.device_data.fan_mode) in AVAILABLE_FAN_MODES:
+            return fan_mode
+        return None
 
     @property
     def fan_modes(self) -> list[str] | None:
         """Return the list of available fan modes."""
-        if self.device_data.fan_modes:
-            return self.device_data.fan_modes
+        if fan_modes := self.device_data.fan_modes:
+            fan_modes_set = set(fan_modes)
+            available_fan_modes = sorted(
```

Comment:
I would assume that fan modes are sorted in a specific order. `sorted` would override that. 

Also, I would calculate this once when the entity is initialized to avoid doing this on every update (since they will never change)

Reply:
Removing the sorting which I agree would not be applicable.
The list is dynamic depending on `hvac_mode` so can't be a set-once.

Reply:
Api is proving the options randomly so need to keep sorted, both for user and for test